### DISCUSSION TITLE
IP Tunnel

### DISCUSSION THREAD

**Discussion by Real-Eragamer** [vpnhood/VpnHood#594] [General]
Hey , i'm using you're app and it's great and its lovely.  i'm having an issue which is ip tunnling.
in my country some isps blocked some ips but there is a trick which we use as a solution, we tunnel a server which is located in our country to the server which is outside of country , and i cant use tunneling because i cant change the ip address i want to connect to. will you help me about it?

**Comment by trudyhood** [comment]
Sure, Many users have already done that.

A contributor did that, but we did not have time to publish it. Please follow the following document. I would appreciate your letting me know if any modifications are required. Also, let me know if the document works or not. It would be great if you could write it in GitHub and share it with me so I can upload it to our wiki page.

https://docs.google.com/document/d/1XKmGY_jcYqaPZ84omw4VF7_SuH3qXT8BzFmHTyFraHw/edit?usp=sharing

I am waiting to hear from you.

**Comment by Real-Eragamer** [comment]
thank for fast responding, and also thanks for helping me. sure i'll let you know when ever i done it

**Comment by trudyhood** [comment]
I took a quick look at it again.
ServerA IP (Your bridge server in your country) should be "PublicInToken" and "IsListen" must be false.
ServerB IP (The VPN Server) should be "Private", and "IsListen" must be yes.

**Reply by Real-Eragamer** [reply]
> I took a quick look at it again. ServerA IP (Your bridge server in your country) should be "PublicInToken" and "IsListen" must be false. ServerB IP (The VPN Server) should be "Private", and "IsListen" must be yes.

thanks , i did test it and it's working great, thank you

**Reply by trudyhood** [reply]
That great! 

- Did you find any issue or challenge to follow that document?
- Do you have any comments or anything to improve the instruction?
- Can you provide a diagram so we can add it to the document and publish it on the VpnHood wiki?

**Reply by Real-Eragamer** [reply]
> That great!
> 
> * Did you find any issue or challenge to follow that document?
> * Do you have any comments or anything to improve the instruction?
> * Can you provide a diagram so we can add it to the document and publish it on the VpnHood wiki?

i was going to read the document but i got a massage from you which you told me to set server A (public token) + no listen and set server B (private) + yes listen and it worked and it was easy. 

i have installed vpn hood in my germany server (server B) and then using iptunneling did tunnel 2 port which were set in pannel (443 and 7248 i think) then i did the private and public and listen part and after that just made a token and then i did connect to it. people can use dokodoor tunnel to , it's better and easy and secure than iptable.

**Reply by trudyhood** [reply]
dokodoor! What is that? Can you let us know how to use tunneling and dokodoor tunnel instead of the iptable? Can you give us that instruction so we can improve that part?

**Reply by Real-Eragamer** [reply]
> dokodoor! What is that? Can you let us know how to use tunneling and dokodoor tunnel instead of the iptable? Can you give us that instruction so we can improve that part?

there are so many ways but i'll tell you a simple way of it. dokodemodoor tunnel , is a way of tunneling which use xray core. there is a team called hiddify and they have made a script for making all tunnels simple.

https://hiddify.com/manager/domain-worker-cdn-and-tunneling/How-to-setup-relay-server-using-Hiddify-Relay-Builder/#installation-and-setup here is their script

**Reply by trudyhood** [reply]
Ok, Thank you

**Comment by Onair-santa** [comment]
**trudyhood**
I really need the ability of the program to work through a SOCKS5 proxy.

**Comment by trudyhood** [comment]
Hi @Onair-santa 
I’m not quite sure what you need. Are you required to connect through a managed SOCKS proxy? VpnHood is a hybrid VPN and can function as a SOCKS proxy in some cases, so you typically don’t need an additional SOCKS proxy—unless your network administrator enforces it.

By the way, I imagine it is a cool feature for places where users enforce to set a proxy in order to access the internet; in this case, we can configure to use the SOCKS proxy so they can bypass that managed SOCKS proxy restriction. If you want this, I will give it to you by the end of this week:

![image](https://github.com/user-attachments/assets/5e2f14f8-1b38-4d23-8c32-3b0c7a4ba35b)

I can't understand your need. Does your network require you to connect to a managed socks proxy? VpnHood is a hybrid VPN and sometimes works like a sock proxy, so you don't need any other sock proxy unless your network administrator forces you.

**Comment by Onair-santa** [comment]
We need the function of the server operation through a connection to the socks proxy on a specific local port of the local proxy program for unblocking our Internet.

**Reply by trudyhood** [reply]
Sorry, but I can't understand the scenario. 
1) Why does the client not directly connect to ByeDPI Plus? Because there is no android client?
2) Why don't you simply run the VpnHoodServer on unblocking network

Please update the diagram and let me know where the firewall is.

**Comment by Onair-santa** [comment]
The situation and work is such that the server must be in a country with a closed Internet. Otherwise, I would have installed servers in Germany and not asked)

**Reply by trudyhood** [reply]
You believe that a SOCKS proxy can bypass filtering (though I still don't know where the firewall is). VpnHood traffic is indistinguishable from ordinary website traffic. If someone could detect VpnHood traffic, other SOCKS-based protocols would stand no chance (perhaps my knowledge is limited).  

Rerouting server traffic to another server using a SOCKS proxy is possible, but it doesn't make much sense.  

It sounds like the VpnHood server is running on a local network and connecting to an unrestricted internet source, such as a Starlink network or some ISP on the same region, which is less restricted and can run SOCKS proxy. If that's the case, there are far better options, such as port forwarding, than using a SOCKS proxy on the server side. Many people are already running VpnHood with that scenario.

By the way, you can describe your scenario to my email, trudy@vpnhood.com

**Comment by Onair-santa** [comment]
OK, I'll write to you later
Thank you for your reply.
